builder/builder.go

```diff
@@ -94,7 +94,7 @@ type Payload struct {
 	NoTxPool  bool
 }
 
-func (b *Builder) Build(payload *Payload) error {
+func (b *Builder) Build(payload *Payload) (*monomer.Block, error) {
```

Comment:
There are some places in tests (maybe they're only in `builder_test.go`) where we call `HeadBlock` right after `Build`. We can clean up those call sites now that we return the block.

Reply:
bd8ce7d